src/Compilers/CSharp/Portable/Errors/MessageProvider.cs

```diff
@@ -103,7 +103,15 @@ public override string ConvertSymbolToString(int errorCode, ISymbol symbol)
                 return symbol.ToString();
             }
 
-            return SymbolDisplay.ToDisplayString(symbol, SymbolDisplayFormat.CSharpShortErrorMessageFormat);
+            var format = SymbolDisplayFormat.CSharpShortErrorMessageFormat;
+
+            // for CS0118 and symbol is a parameter, we dont want to display the type, but the name
+            if (errorCode == (int)ErrorCode.ERR_BadSKknown)
+            {
+                format = format.WithParameterOptions(SymbolDisplayParameterOptions.IncludeName);
```

Comment:
How about instead adding an overload with the formatting options and then change the call site to pass the IncludeName option?